### ISSUE TITLE
[BUG] Whoogle crashes upon startup in Docker

### ISSUE THREAD

**Issue by spike-the-coder** [benbusby/whoogle-search#5]
**Describe the bug**
Whoogle crashes immediately in Docker after running the setup commands.

**To Reproduce**
1. Enter in your terminal:
```
git clone https://github.com/benbusby/whoogle-search.git
cd whoogle-search
docker build --tag whooglesearch:1.0 .
docker run --publish 8888:5000 --detach --name whooglesearch whooglesearch:1.0
```
2. Type `docker ps -a` and see:
```
CONTAINER ID        IMAGE                                    COMMAND                  CREATED             STATUS                       PORTS                                      NAMES
d05696f259ef        whooglesearch:1.0                        "./whoogle-search"       21 seconds ago      Exited (137) 2 seconds ago                                              whooglesearch
```
3. View logs at `docker logs whooglesearch` and see error:
```
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
./whoogle-search: line 27:    10 Killed                  flask run --host="0.0.0.0" --port=$PORT
```

**Expected behavior**
Whoogle should be able to start in Docker without crashing.

**Screenshots**
N/A

**Environment:**
- Hosting: DigitalOcean
- Image: [Docker Droplet](https://marketplace.digitalocean.com/apps/docker)
- OS: Ubuntu 18.04

**Additional context**
N/A

**Comment by CHBMB** [comment]
I can't reproduce this, I managed to build it and get it up and running with:

Git clone
```
git clone https://github.com/benbusby/whoogle-search.git
```

Build logs
```
chbmb@thinkpad:~/Downloads/whoogle-search-master$ docker build -t chbmb/whoogle .
Sending build context to Docker daemon  685.1kB
Step 1/6 : FROM python:3
3: Pulling from library/python
90fe46dd8199: Pull complete 
35a4f1977689: Pull complete 
bbc37f14aded: Pull complete 
74e27dc593d4: Pull complete 
4352dcff7819: Pull complete 
deb569b08de6: Pull complete 
98fd06fa8c53: Pull complete 
7b9cc4fdefe6: Pull complete 
512732f32795: Pull complete 
Digest: sha256:ad7fb5bb4770e08bf10a895ef64a300b288696a1557a6d02c8b6fba98984b86a
Status: Downloaded newer image for python:3
 ---> 4f7cd4269fa9
Step 2/6 : WORKDIR /usr/src/app
 ---> Running in af5118e31093
Removing intermediate container af5118e31093
 ---> 885170d3fc29
Step 3/6 : COPY . .
 ---> 53ed7cd75eee
Step 4/6 : RUN pip install --no-cache-dir -r requirements.txt
 ---> Running in 430d3f51d88a
Collecting beautifulsoup4==4.8.2
  Downloading beautifulsoup4-4.8.2-py3-none-any.whl (106 kB)
Collecting bs4==0.0.1
  Downloading bs4-0.0.1.tar.gz (1.1 kB)
Collecting cffi==1.13.2
  Downloading cffi-1.13.2-cp38-cp38-manylinux1_x86_64.whl (439 kB)
Collecting Click==7.0
  Downloading Click-7.0-py2.py3-none-any.whl (81 kB)
Collecting cryptography==2.8
  Downloading cryptography-2.8-cp34-abi3-manylinux2010_x86_64.whl (2.3 MB)
Collecting Flask==1.1.1
  Downloading Flask-1.1.1-py2.py3-none-any.whl (94 kB)
Collecting itsdangerous==1.1.0
  Downloading itsdangerous-1.1.0-py2.py3-none-any.whl (16 kB)
Collecting Jinja2==2.10.3
  Downloading Jinja2-2.10.3-py2.py3-none-any.whl (125 kB)
Collecting MarkupSafe==1.1.1
  Downloading MarkupSafe-1.1.1-cp38-cp38-manylinux1_x86_64.whl (32 kB)
Collecting Phyme==0.0.9
  Downloading Phyme-0.0.9.tar.gz (1.3 MB)
Collecting pycparser==2.19
  Downloading pycparser-2.19.tar.gz (158 kB)
Collecting pycurl==7.43.0.4
  Downloading pycurl-7.43.0.4.tar.gz (215 kB)
Collecting pyOpenSSL==19.1.0
  Downloading pyOpenSSL-19.1.0-py2.py3-none-any.whl (53 kB)
Collecting pytest==5.4.1
  Downloading pytest-5.4.1-py3-none-any.whl (246 kB)
Collecting python-dateutil==2.8.1
  Downloading python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
Collecting six==1.14.0
  Downloading six-1.14.0-py2.py3-none-any.whl (10 kB)
Collecting soupsieve==1.9.5
  Downloading soupsieve-1.9.5-py2.py3-none-any.whl (33 kB)
Collecting Werkzeug==0.16.0
  Downloading Werkzeug-0.16.0-py2.py3-none-any.whl (327 kB)
Collecting pluggy<1.0,>=0.12
  Downloading pluggy-0.13.1-py2.py3-none-any.whl (18 kB)
Collecting more-itertools>=4.0.0
  Downloading more_itertools-8.2.0-py3-none-any.whl (43 kB)
Collecting wcwidth
  Downloading wcwidth-0.1.9-py2.py3-none-any.whl (19 kB)
Collecting packaging
  Downloading packaging-20.3-py2.py3-none-any.whl (37 kB)
Collecting py>=1.5.0
  Downloading py-1.8.1-py2.py3-none-any.whl (83 kB)
Collecting attrs>=17.4.0
  Downloading attrs-19.3.0-py2.py3-none-any.whl (39 kB)
Collecting pyparsing>=2.0.2
  Downloading pyparsing-2.4.7-py2.py3-none-any.whl (67 kB)
Building wheels for collected packages: bs4, Phyme, pycparser, pycurl
  Building wheel for bs4 (setup.py): started
  Building wheel for bs4 (setup.py): finished with status 'done'
  Created wheel for bs4: filename=bs4-0.0.1-py3-none-any.whl size=1272 sha256=99eb86407f52f07af2fa614bcd4ac86f894cd74dabefbdae9734d94cd95d52fe
  Stored in directory: /tmp/pip-ephem-wheel-cache-svr78e92/wheels/75/78/21/68b124549c9bdc94f822c02fb9aa3578a669843f9767776bca
  Building wheel for Phyme (setup.py): started
  Building wheel for Phyme (setup.py): finished with status 'done'
  Created wheel for Phyme: filename=Phyme-0.0.9-py3-none-any.whl size=1379058 sha256=bed58c36b0581319036568e51bf675cd19b315c1d16bbe0875e6201b1997d89a
  Stored in directory: /tmp/pip-ephem-wheel-cache-svr78e92/wheels/1e/1b/46/ca37019ef47f5b3adc662675f4f33b6eed706a8cae95c641c2
  Building wheel for pycparser (setup.py): started
  Building wheel for pycparser (setup.py): finished with status 'done'
  Created wheel for pycparser: filename=pycparser-2.19-py2.py3-none-any.whl size=111031 sha256=16904d5ca6a867ccb2860fb6cc70a9e86b620074375d6c9d6c91ac2cdf9eca29
  Stored in directory: /tmp/pip-ephem-wheel-cache-svr78e92/wheels/6d/fb/fb/c752da1378a60304d18004cc5c58e73519a798ee2809db7562
  Building wheel for pycurl (setup.py): started
  Building wheel for pycurl (setup.py): finished with status 'done'
  Created wheel for pycurl: filename=pycurl-7.43.0.4-cp38-cp38-linux_x86_64.whl size=313842 sha256=6ae6d4c59b6db6d67ff4c88c5898bc571adb1be91f3bee59356ea530a012f9e1
  Stored in directory: /tmp/pip-ephem-wheel-cache-svr78e92/wheels/25/7d/08/2c70c9edaef57630f847325b3d58056cb2eb065d89fb46b4d9
Successfully built bs4 Phyme pycparser pycurl
Installing collected packages: soupsieve, beautifulsoup4, bs4, pycparser, cffi, Click, six, cryptography, itsdangerous, MarkupSafe, Jinja2, Werkzeug, Flask, Phyme, pycurl, pyOpenSSL, pluggy, more-itertools, wcwidth, pyparsing, packaging, py, attrs, pytest, python-dateutil
Successfully installed Click-7.0 Flask-1.1.1 Jinja2-2.10.3 MarkupSafe-1.1.1 Phyme-0.0.9 Werkzeug-0.16.0 attrs-19.3.0 beautifulsoup4-4.8.2 bs4-0.0.1 cffi-1.13.2 cryptography-2.8 itsdangerous-1.1.0 more-itertools-8.2.0 packaging-20.3 pluggy-0.13.1 py-1.8.1 pyOpenSSL-19.1.0 pycparser-2.19 pycurl-7.43.0.4 pyparsing-2.4.7 pytest-5.4.1 python-dateutil-2.8.1 six-1.14.0 soupsieve-1.9.5 wcwidth-0.1.9
Removing intermediate container 430d3f51d88a
 ---> 841f7318e047
Step 5/6 : RUN chmod +x ./whoogle-search
 ---> Running in 30018275336f
Removing intermediate container 30018275336f
 ---> 985fb971248d
Step 6/6 : CMD ["./whoogle-search"]
 ---> Running in c3e4f705c499
Removing intermediate container c3e4f705c499
 ---> 665f4b02db5e
Successfully built 665f4b02db5e
Successfully tagged chbmb/whoogle:latest
```

Docker Run
```
docker run -d --name=whoogle \
-p '5000:5000/tcp \
chbmb/whoogle
```

Logs
```
* Environment: production
WARNING: This is a development server. Do not use it in a production deployment.
Use a production WSGI server instead.
* Debug mode: off
* Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)
```

**Comment by benbusby** [comment]
I just built the project on the same environment (docker droplet on DO) and didn't encounter any issues. 

That being said, the version I built also included the fix provided in #13 where the Dockerfile was updated with a specific call to expose port 5000 before running, so that could be why I'm not encountering any issues as well. @spike-the-coder would you mind trying with the latest version of the repo to see if you're still encountering any issues?

**Comment by spike-the-coder** [comment]
Sorry for the confusion, I realized my droplet didn't have enough memory to run my current docker containers + Whoogle and so Docker was automatically killing Whoogle whenever I tried to run it. Thanks for your assistance though!
